File: Algorithm/Programers_덧칠하기.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(int n, int m, vector<int> section) {
	int answer = 0;
	vector<bool> wall(n + 1, true);
	for (auto iter : section)
		wall[iter] = false;

	for (int i = 1; i <= n; i++) {
		if (wall[i]) continue;
		answer++;
		for (int j = 1; j < m; j++) {
			if (i > n) break;
			wall[i++] = true;
		}

	}

	return answer;
}
//https://school.programmers.co.kr/learn/courses/30/lessons/161989

File: Algorithm/Programers_덧칠하기.java
import java.util.*;
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        int notPaintedPos =0;
        for (int pos : section) {
            if(notPaintedPos>pos) continue;
            notPaintedPos = pos + m;
            answer++;
        }
        return answer;
    }
}
class main{
    public static void main(String[] args) {
        Solution sol = new Solution();
        int[] arr = {2, 3, 6};
        System.out.println("sol = " + sol.solution(8,4,arr));
    }
}
//https://school.programmers.co.kr/learn/courses/30/lessons/161989